sqlmesh/core/loader.py

```diff
@@ -386,11 +386,18 @@ def _load_models(
         Loads all of the models within the model directory with their associated
         audits into a Dict and creates the dag
         """
-        models = self._load_sql_models(macros, jinja_macros, audits, signals)
-        models.update(self._load_external_models(audits, gateway))
-        models.update(self._load_python_models(macros, jinja_macros, audits, signals))
+        sql_models = self._load_sql_models(macros, jinja_macros, audits, signals)
+        external_models = self._load_external_models(audits, gateway)
+        python_models = self._load_python_models(macros, jinja_macros, audits, signals)
 
-        return models
+        all_model_names = (
+            list(sql_models.keys()) + list(external_models.keys()) + list(python_models.keys())
+        )
+        duplicates = [name for name, count in t.Counter(all_model_names).items() if count > 1]
```

Comment:
I believe we should be importing `Counter` from `collections`, because `typing.Counter` is [deprecated](https://docs.python.org/3/library/typing.html#typing.Counter).